pkg/repo_add/repoadd.go

```diff
@@ -0,0 +1,150 @@
+// Package repo_add provides a wrapper around the
+// repo_add program.
+package repo_add
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+const repoAddCommand string = "repo-add"
+const repoRemoveCommand string = "repo-remove"
+
+// CommonOptions are options that are common to the repo-add and repo-remove commands.
+// Derived from https://www.archlinux.org/pacman/repo-add.8.html
+type CommonOptions struct {
+	// Generate a PGP signature file using GnuPG.
+	// This will execute gpg --detach-sign --use-agent on the generated database to generate
+	// a detached signature file, using the GPG agent if it is available.
+	// The signature file will be the entire filename of the database with a “.sig” extension.
+	// equivalent to `repo-add --sign`
+	Sign bool
```

Comment:
Do we want to sign through repo-add or do the signing ourselves? As doing it throug repo-add means interfacing with the systems gpg which may or may not be what we want

Reply:
We probably want do do it ourselves. I didn't just include it for us to actually use it but mostly just to be complete and actually implement the entire command line interface